reana_workflow_engine_snakemake/executor.py

```diff
@@ -177,13 +184,36 @@ def handle_job_success(self, job):
             job, upload_remote=False, handle_log=False, handle_touch=True
         )
 
-        self._handle_job_status(job, "finished")
+        self._handle_workflow_status(job, RunStatus.running)
 
     def handle_job_error(self, job):
         """Override job error method to publish job status."""
         super().handle_job_error(job)
 
-        self._handle_job_status(job, "failed")
+        self._handle_workflow_status(job, RunStatus.failed)
+
+    def _get_job_status_from_controller(self, job_id: str) -> str:
+        response = self.rjc_api_client.check_status(job_id)
+        return response["status"]
+
+    def _poll_jobs(self):
+        while True:
+            with self.lock:
+                active_jobs = self.active_jobs
+
+            log.debug(f"Active jobs: {self.active_jobs}")
+            for active_job in active_jobs:
+                job_id = active_job.job.reana_job_id
+                status = self._get_job_status_from_controller(job_id)
+                if status == JobStatus.finished:
+                    self.handle_job_success(active_job.job)
+                elif status == JobStatus.failed:
+                    self.handle_job_error(active_job.job)
```

Comment:
*question:* Should we care about `stopped` and other statuses here?

Reply:
Does +1 mean `yes`?

Reply:
I tested `stopped` without any new changes, and it worked. So, I assume no need to account for that here as I haven't seen any functions that handle `stopped` status in some other workflow engines.

Reply:
Have you checked how is it handled then? Can't remember from the top of my head.

Reply:
The call is issued to `r-w-controller` to set `stop` status ([code here](https://github.com/reanahub/reana-workflow-controller/blob/a3e2af49a4c8153a46cee939ff74cfb81ae2c356/reana_workflow_controller/rest/utils.py#L119)). It will delete batch pods and all running jobs. Acts as a force stop, no graceful shutdown yet.